receiver/mongodbreceiver/integration_test.go

```diff
@@ -137,9 +142,13 @@ func TestMongodbIntegration(t *testing.T) {
 	}
 
 	for _, tt := range testCases {
+		// Without tt := tt, tt is captured by the anonymous function passed to t.Run() in the for loop.
+		// This can lead to unexpected behavior because the anonymous function may not execute immediately, which means that the
+		// value of tt may have changed by the time the function is actually executed.
+		tt := tt
```

Comment:
I think if we define `func (tt testCase) run(t *testing.T)`, we won't need such an explanation. WDYT?

Reply:
Works and looks much nicer.